src/node_shell_read.js

```diff
@@ -11,7 +11,10 @@ read_ = (filename, binary) => {
     return binary ? ret : ret.toString();
   }
 #endif
-  filename = nodePath['normalize'](filename);
+  // We need to re-wrap `file://` strings to URLs. Normalizing isn't
```

Comment:
Wait, why? The string is already a URL, you don't need to create an `URL` object (which Node.js will convert once again to string anyway).

Reply:
I think you can actually remove this line altogether - normalize (that was here before your change) shouldn't be necessary either, as `readFileSync` happily accepts non-normalized / relative URLs too.

Reply:
So `filename`, whatever it is - whether a URL string or file path - can just be passed unmodified to `readFileSync`.

Reply:
IIUC, the `.href` property (formerly the `.toString()` call) when the `wasmBinaryFile` is assigned made this a `string`.

Reply:
Huh :/ I was sure it accepts file URLs as strings and that I used it as such... Sorry for misleading.